src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts

```diff
@@ -210,47 +230,22 @@ export class RawSqlResultsToEntityTransformer {
         metadata: EntityMetadata,
     ): boolean {
         let hasData = false
-        metadata.columns.forEach((column) => {
-            // if table inheritance is used make sure this column is not child's column
-            if (
-                metadata.childEntityMetadatas.length > 0 &&
-                metadata.childEntityMetadatas.findIndex(
-                    (childMetadata) => childMetadata.target === column.target,
-                ) !== -1
-            )
-                return
-
-            const value =
-                rawResults[0][
-                    DriverUtils.buildAlias(
-                        this.driver,
-                        undefined,
-                        alias.name,
-                        column.databaseName,
-                    )
-                ]
-            if (value === undefined || column.isVirtual) return
+        const result = rawResults[0]
+        for (const [key, column] of this.getColumnsToProcess(
```

Comment:
Maybe we should rename `key` as `columnAlias`?
It would make the code a bit more explicit since the method name `getColumnsToProcess()` is not indicating what `key` is.